PolyEngine/Engine/Src/InputWorldComponent.cpp

```diff
@@ -17,3 +17,100 @@ bool Poly::InputWorldComponent::IsPressed(const std::initializer_list<eMouseButt
 		result = result && CurrMouseButton[i];
 	return result;
 }
+
+bool Poly::InputWorldComponent::IsPressed(size_t controllerID, eControllerButton button) const
+{
+	if(ControllerPointers.GetSize() <= controllerID)
+	{
+		return false;
+	}
+	SDL_GameController* controller = ControllerPointers[controllerID];
+	if(controller != nullptr)
+	{
+		return Controllers.at(controller).CurrButton[button];
+	}
+	else
+	{
+		return false;
+	}
+}
+
+bool Poly::InputWorldComponent::IsClicked(size_t controllerID, eControllerButton button) const
+{
+	if(ControllerPointers.GetSize() <= controllerID)
```

Comment:
This is candidate for assertion in each of these functions. Just check `ASSERTE(IsControllerConnected(idx), "Controller not connected!");`